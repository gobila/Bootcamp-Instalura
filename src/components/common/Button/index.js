import React from 'react';
import styled, { css } from 'styled-components';
import get from 'lodash/get';
import PropTypes from 'prop-types';
import { TextStyleVariantsMap } from '../../foundation/Text';
import { breakpointsMedia } from '../../theme/utils/breakpointsMedia';
import { propsToStyle } from '../../theme/utils/propsToStyle';
import Link from '../Link';

const ButtonGhost = css`
    color: ${(props) => get(props.theme, `colors.${props.variant}.color`)};
    background-color: transparent;
`;
const ButtonDefault = css`
    color:${(props) => get(props.theme, `colors.${props.variant}.contrastText`)};
    background-color: ${(props) => get(props.theme, `colors.${props.variant}.color`)};
`;

const ButtonWrapper = styled.button`
    border: 0;
    cursor: pointer;
    padding: 12px 26px;
    font-weight: bold;
    opacity: 1;
    border-radius: 8px;
    /* ${TextStyleVariantsMap.smallestException} */
    /* color: white;
    background-color: #D7385E; */
    ${function BtnProps(props) {
    if (props.ghost) {
      return ButtonGhost;
    }
    return ButtonDefault;
  }}
    transition: opacity${({ theme }) => theme.transition};
    border-radius: ${({ theme }) => theme.borderRadius};
    &:hover,
    &:focus {
        opacity: .5;
    }

    ${breakpointsMedia({
    xs: css`
            ${TextStyleVariantsMap.smallestException}
            padding: 8px 16px;
        `,
    md: css`
            ${TextStyleVariantsMap.paragraph1}
        `,
  })}
    &:disabled {
        cursor: not-allowed;
        opacity: .2;
    }
    ${({ fullWidth }) => fullWidth && css`
        width: 100%;
    `};

    ${propsToStyle('margin')}
    ${propsToStyle('display')}
`;

export function Button({ href, children, ...props }) {
  const hasHref = Boolean(href);
  const tag = hasHref ? Link : 'button';
  return (
    <ButtonWrapper
      as={tag}
      href={href}
      {...props}
    >
      {children}
    </ButtonWrapper>
  );
}

Button.defaultProps = {
  href: undefined,
};

Button.propTypes = {
  href: PropTypes.string,
  children: PropTypes.node.isRequired,
};
