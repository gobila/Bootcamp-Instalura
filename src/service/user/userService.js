import { isStagingEnv } from '../../infra/env/isStagingEnv';
import { HTTPClient } from '../../infra/http/HTTPClient';
import { authService } from '../auth/authService';

// VALE A PENA CRIAR UM NOVO ARQUIVO SO PARA ESSA URL
// API para o login
// const apiURL = 'https://instalura-api-git-master-omariosouto.vercel.app/api/login';
const BASE_URL = isStagingEnv
// Back end de Dev
  ? 'https://instalura-api-git-master-omariosouto.vercel.app'
// Back end de prod
  : 'https://instalura-api.vercel.app';

export const userService = {
  async getProfilePage(ctx) {
    const url = `${BASE_URL}/api/users/posts/?per_page=10&order=DESC`;
    try {
      const token = await authService(ctx).getToken();
      const response = await HTTPClient(url, {
        headers: {
          Authorization: `Bearer ${token}`,
        },
      });
      return {
        user: {
          totalLikes: 1,
        },
        posts: response.data,
      };
    } catch (err) {
      throw new Error('Nao conseguiu trazer os post do user', err);
    }
  },

  // New Post
  async newPost(data) {
    const url = `${BASE_URL}/api/posts`;
    try {
      const token = await authService().getToken();
      const response = await HTTPClient(url, {
        method: 'POST',
        headers: {
          Authorization: `Bearer ${token}`,
        },
        body: data,
      });
      return response.data;
    } catch (err) {
      throw new Error('NÃ£o foi possivel enviar o novo post', err);
    }
  },

  // GitHub infos
  async githubInfos(user) {
    const url = `https://api.github.com/users/${user}`;
    // passando o objeto vazio pois HTTPClient() requer 2 argumentos sendo o segundo as
    // opÃ§oes de hearder e body
    const response = await HTTPClient(url, { });
    return response;
    // const restona = await fetch('https://api.github.com/users/omariosouto')
    //   .then((res) => res.json())
    //   .then((resp) => resp);
    // return restona
  },
};
