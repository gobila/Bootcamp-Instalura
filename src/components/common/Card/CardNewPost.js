import React, { useState } from 'react';
import styled, { css } from 'styled-components';
import { Box } from '../../foundation/layout/Box';
import Text from '../../foundation/Text';
import FormPost from '../../patterns/FormPost';
import { breakpointsMedia } from '../../theme/utils/breakpointsMedia';
import { Button } from '../Button';

const CardWrapper = styled.div`
  max-width: 600px;
  margin: 25px auto 25px auto;
  display:flex;
  flex-direction: column;
  background-color:${({ theme }) => (theme.colors.background.light.color)};
  ${breakpointsMedia({
    xs: css`
        max-width: 90%;
        `,
    sm: css`
        max-width: 576px; 
        `,
  })}
`;
const CardHead = styled(Box)`
  width: 600px;
  display: inherit;
  align-items:center;
  padding: 25px 40px;
  img{
    width:50px;
    height: 50px;
    border-radius: 25px;
    margin-right:19px
  }
`;
const CardBody = styled(Box)`
  display:flex;
  flex-direction: column;
  max-width: 100%;
  max-height: 650px;
  img{
    width:100%;
    object-fit: fill;
  }
`;
const CardFooter = styled(Box)`
  margin: 10px 25px;
  max-width: 600px;
  display: flex;
  flex-direction:column;
  justify-content: space-between;
  img{
    width:150px;
    height: 150px;
    /* border-radius: 25px; */
    margin-right:auto
  }
  input + label{
    cursor: pointer;
  }
  input[type="radio"]:checked + label{
    border: red 5px solid;
  }
`;

export function CardNewPost({
  propsDoModal,
}) {
  const [image, setImage] = useState('');
  const [filtro, setFiltro] = useState('');

  return (
    <CardWrapper {...propsDoModal}>
      <CardHead justifyContent="flex-end">
        <Button
          ghost
          data-modal-btn-close="true"
        >
          ❌
        </Button>
      </CardHead>
      <CardBody>
        <Box maxHeight="500px" display="flex" marginBottom="10px">
          <img
            className={filtro}
            src={image || 'https://place-hold.it/550/9c27b/fffff&text=%F0%9F%99%8E%E2%80%8D%E2%99%82%EF%B8%8F'}
            alt=""
          />
        </Box>

        {/* <form style={{ width: '90%', display: 'flex', margin: 'auto' }}>
          <TextField
            placeholder="URL da Imagem"
            name="imageURL"
            value={imageURL}
            onChange={handleChange}
            previewPhoto
          />
          <Button
            variant="secondary.main"
            onClick={(e) => {
              e.preventDefault();
              return setImage(imageURL);
            }}
            style={{ borderRadius: '0px 9px 9px 0px', width: '10%' }}
          >
            ➡
          </Button>
        </form> */}
        <Text
          variant="smallestException"
          textAlign="center"
          margin="5px"
        >
          Formatos suportados: jpg, png, svg e xpto.
        </Text>
      </CardBody>
      <CardFooter>
        <FormPost
          onSubmit={(e) => {
            e.preventDefault();
            setImage(e.target.photoUrl.value);
          }}
          photoOk={image}
          filterList={(e) => setFiltro(e.target.value)}
        />
      </CardFooter>
    </CardWrapper>
  );
}
