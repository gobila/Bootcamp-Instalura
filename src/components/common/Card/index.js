import React from 'react';
import styled from 'styled-components';
import { Dots } from '../../../assets/dots';
import { Box } from '../../foundation/layout/Box';
import Text from '../../foundation/Text';
import { Button } from '../Button';

export { CardNewPost } from './CardNewPost';

const CardWrapper = styled.div`
  max-width: 600px;
  margin: 25px auto 25px auto;
  display:flex;
  flex-direction: column;
  background-color:${({ theme }) => (theme.colors.background.light.color)};
`;
const CardHead = styled(Box)`
  max-width: 600px;
  display: inherit;
  justify-content: space-between;
  align-items:center;
  padding: 25px 40px;
  img{
    width:50px;
    height: 50px;
    border-radius: 25px;
    margin-right:19px
  }
`;
const CardBody = styled(Box)`
  max-width: 100%;
  max-height: 600px;
  img{
    width:100%;
  }
`;
const CardFooter = styled(Box)`
  margin: 10px 25px;
  max-width: 600px;
  display: flex;
  flex-direction:column;
  justify-content: space-between;
  img{
    width:30px;
    height: 30px;
    border-radius: 25px;
    margin-right:19px
  }
`;

export default function Card({
  Post, UserName, likes, filter, description,
}) {
  const avatar = 'https://upload.wikimedia.org/wikipedia/commons/c/c0/Nicolas_Cage_Deauville_2013.jpg';
  return (
    <CardWrapper>
      <CardHead>
        <Box
          display="flex"
          alignItems="center"
        >
          <img src={avatar} alt="Nicolas Cage" />
          <Text tag="p" variant="paragraph1">
            {UserName}
          </Text>
        </Box>
        <Button icon>
          {/* <img src="/images/assets/iconMoreLight.svg" alt="" /> */}
          <Dots />
        </Button>
      </CardHead>
      <CardBody>
        <img src={Post} className={filter} alt={description} />
      </CardBody>
      <CardFooter>
        <Box display="flex" justifyContent="space-between">
          <Box display="flex" alignItems="center">
            <Button icon>
              <img src="/images/assets/heart1.png" alt="Curtidas" />
            </Button>
            {likes}
            <Button icon>
              <img src="/images/assets/message-circle.png" alt="comentarios" />
            </Button>
            {likes}
            <Button icon>
              <img src="/images/assets/send.png" alt="compartilhar" />
            </Button>
          </Box>
          <Button icon>
            <img src="/images/assets/bookmark.png" alt="salvar" />
          </Button>
        </Box>
        <Box display="flex" alignItems="center">
          <img src={avatar} alt="Nicolas Cage" />
          <Text tag="p" variant="comments">
            So excited to play this new...
          </Text>
          <Button comments>
            <Text tag="p">Mais</Text>
          </Button>
        </Box>
      </CardFooter>
    </CardWrapper>
  );
}
