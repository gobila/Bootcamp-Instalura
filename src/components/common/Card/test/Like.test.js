import React from 'react';
import user from '@testing-library/user-event';
import { render, screen } from '../../../../infra/test/testUtils';
import Card from '../index';
import { ButtonLike } from '../../Button';
// Post, user, allLikes, filter, description, avatarURL, imgId,
describe('<Card /> like option', () => {
  // eslint-disable-next-line jest/expect-expect
  test('renders components', () => {
    render(
      <Card
        Post="/images/bubbles.svg"
        user="omariosouto"
        allLikes="1"
        filter="none"
        description="description"
        avatarURL="/images/bubbles.svg"
        imgId="1"
      />,
    );
    // screen.debug();
    expect(Card).toMatchSnapshot();
  });
  describe('when click in like button', () => {
    test('clicked', () => {
      const onClickMock = jest.fn();
      render(
        <ButtonLike
          liked="true"
          name="Curtidas"
          data-imgId="imgId"
          imgId="1234"
          handleLike={onClickMock}
        />,
      );
      screen.debug();
      const btn = screen.getByRole('button');
      user.click(btn);
      expect(onClickMock).toHaveBeenCalledTimes(1);
    });
  });
});
