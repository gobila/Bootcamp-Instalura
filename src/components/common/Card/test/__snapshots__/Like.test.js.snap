// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> like option renders components 1`] = `[Function]`;
